core/src/main/scala/caliban/HttpRequestMethod.scala

```diff
@@ -0,0 +1,23 @@
+package caliban
+
+import caliban.CalibanError.ValidationError
+import zio.{ FiberRef, Scope, UIO, URIO, Unsafe, ZIO }
+
+private[caliban] sealed trait HttpRequestMethod
+
+private[caliban] object HttpRequestMethod {
+  case object GET  extends HttpRequestMethod
+  case object POST extends HttpRequestMethod
+
+  val MutationOverGetError: ValidationError = ValidationError("Mutations are not allowed for GET requests", "")
```

Comment:
Btw seeing this make me thing we should probably extend `NoStackTrace` for `ValidationError` at least. What do you think?

Reply:
Quite possibly yes. Actually I think we should probably do that with all the errors since we override `getCause` in `ParsingError` and `ExecutionError`, and the stacktrace of the cause is included in the reported error.

The only downside to that is that `getCause` will now be `null` for `ValidationError`, which might cause some some NPEs to users if they've been reckless with it. I think that should be fine though as long as we point it out in the release notes. 

Also, if one really needs the stack traces, they can set the `scala.control.noTraceSuppression` system property to `true`, which effectively makes `NoStackTrace` to behave like a normal `Throwable`.

I can make the changes in a followup PR

Reply:
Sounds good!